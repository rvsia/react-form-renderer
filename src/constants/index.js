export const components = {
  TEXT_FIELD: 'text-field',
  TEXTAREA_FIELD: 'textarea-field',
  FIELD_ARRAY: 'field-array',
  SELECT_COMPONENT: 'select-field',
  FIXED_LIST: 'fixed-list',
  CHECKBOX: 'checkbox',
  SUB_FORM: 'sub-form',
  RADIO: 'radio',
  TABS: 'tabs',
  TAB_ITEM: 'tab-item',
  DATE_PICKER: 'date-picker',
  TIME_PICKER: 'time-picker',
  TAG_CONTROL: 'tag-control',
  WIZARD: 'wizard',
};
  /** Validator functions placeholder */
export const validators = {
  REQUIRED: 'required-validator',
  /**
     * min length of the input value
     */
  MIN_LENGTH: 'min-length-validator',
  /**
     * max length of the input value
     */
  MAX_LENGTH: 'max-length-validator',
  /**
   * exact length of input value
   */
  EXACT_LENGTH: 'exact-length-validator',
  /**
     * minimum count of fileds in some dynamic list of fields
     */
  MIN_ITEMS_VALIDATOR: 'min-items-validator',
  /**
     * Minimum value of number input
     */
  MIN_NUMBER_VALUE: 'min-number-value',
  /**
     * Maximum value of number inpuy
     */
  MAX_NUMBER_VALUE: 'max-number-value',
  /**
     * Regexp pattern validator
     */
  PATTERN_VALIDATOR: 'pattern-validator',
};

export const layoutComponents = {
  FORM_WRAPPER: 'FormWrapper',
  BUTTON: 'Button',
  COL: 'Col',
  FORM_GROUP: 'FormGroup',
  BUTTON_GROUP: 'ButtonGroup',
  ICON: 'Icon',
  ARRAY_FIELD_WRAPPER: 'ArrayFieldWrapper',
  HELP_BLOCK: 'HelpBlock',
};
