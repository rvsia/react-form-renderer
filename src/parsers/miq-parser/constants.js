import { components } from '../../constants';

export const componentMap = {
  DialogFieldTextBox: components.TEXT_FIELD,
  DialogFieldRadioButton: components.RADIO,
  DialogFieldCheckBox: components.CHECKBOX,
  DialogFieldTextAreaBox: components.TEXTAREA_FIELD,
  DialogFieldDropDownList: components.SELECT_COMPONENT,
  DialogFieldDateControl: components.DATE_PICKER,
  DialogFieldDateTimeControl: components.TIME_PICKER,
  DialogFieldTagControl: components.TAG_CONTROL,
};

// for true values
export const neededAttributes = [
  'name',
  'label',
  [ 'label', 'title' ],
  [ 'data_type', 'dataType' ],
  [ 'required', 'isRequired' ],
  [ 'read_only', 'isReadOnly' ],
  [ 'description', 'helperText' ],
];

// for false values
export const neededFalseAttributes = [
  [ 'visible', 'isVisible' ],
];
