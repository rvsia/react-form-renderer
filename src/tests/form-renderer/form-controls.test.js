import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import FormControls from '../../form-renderer/form-controls';
import { layoutComponents } from '../../constants';
import RendererContext, { configureContext } from '../../form-renderer/renderer-context';

describe('<FormControls />', () => {
  let initialProps;
  let ContextWrapper;
  let initialLayout;
  beforeEach(() => {
    initialLayout = {
      [layoutComponents.BUTTON]: ({ label, ...props }) => <button { ...props }>{ label }</button>,
      [layoutComponents.COL]: ({ children, ...props }) => <div { ...props }>{ children }</div>,
      [layoutComponents.FORM_GROUP]: ({ children, ...props }) => <div { ...props }>{ children }</div>,
      [layoutComponents.BUTTON_GROUP]: ({ children, ...props }) => <div { ...props }>{ children }</div>,
    };
    ContextWrapper = ({ children, ...props }) => (
      <RendererContext.Provider
        value={ configureContext({ layoutMapper: initialLayout, ...props }) }
      >
        { children }
      </RendererContext.Provider>
    );
    initialProps = {
      onSubmit: jest.fn(),
      onReset: jest.fn(),
      onCancel: jest.fn(),
      canReset: true,
      pristine: true,
    };
  });

  it('should render all controls and with default labels', () => {
    const wrapper = mount(
      <ContextWrapper>
        <FormControls { ...initialProps }  disableSubmit/>
      </ContextWrapper>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.find('button').first().simulate('click');
    wrapper.find('button').at(1).simulate('click');
    wrapper.find('button').last().simulate('click');

    expect(initialProps.onSubmit).not.toHaveBeenCalled();
    expect(initialProps.onReset).not.toHaveBeenCalled();
    expect(initialProps.onCancel).toHaveBeenCalled();
  });

  it('should render only submit button', () => {
    const wrapper = mount(
      <ContextWrapper>
        <FormControls { ...initialProps } canReset={ false } onCancel={ undefined } />
      </ContextWrapper>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should enable reset button if pristine is false', () => {
    const wrapper = mount(
      <ContextWrapper>
        <FormControls { ...initialProps } pristine={ false } />
      </ContextWrapper>
    );

    wrapper.find('button').at(1).simulate('click');
    expect(initialProps.onReset).toHaveBeenCalled();
  });

  it('should enable submit button if canSubmit is true', () => {
    const wrapper = mount(
      <ContextWrapper>
        <FormControls { ...initialProps } canSubmit />
      </ContextWrapper>
    );

    wrapper.find('button').first().simulate('click');
    expect(initialProps.onSubmit).toHaveBeenCalled();
  });
});
