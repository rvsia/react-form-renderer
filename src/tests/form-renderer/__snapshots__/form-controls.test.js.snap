// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormControls /> should render all controls and with default labels 1`] = `
<ContextWrapper>
  <FormControls
    canReset={true}
    canSubmit={false}
    cancelLabel="Cancel"
    disableSubmit={true}
    onCancel={[MockFunction]}
    onReset={[MockFunction]}
    onSubmit={[MockFunction]}
    pristine={true}
    resetLabel="Reset"
    submitLabel="Submit"
  >
    <Component
      xs={12}
    >
      <div
        xs={12}
      >
        <Component>
          <div>
            <Component>
              <div>
                <Component
                  disabled={true}
                  label="Submit"
                  onClick={[MockFunction]}
                  type="button"
                  variant="primary"
                >
                  <button
                    disabled={true}
                    onClick={[MockFunction]}
                    type="button"
                    variant="primary"
                  >
                    Submit
                  </button>
                </Component>
                <Component
                  disabled={true}
                  label="Reset"
                  onClick={[MockFunction]}
                  type="button"
                >
                  <button
                    disabled={true}
                    onClick={[MockFunction]}
                    type="button"
                  >
                    Reset
                  </button>
                </Component>
                <Component
                  label="Cancel"
                  onClick={[MockFunction]}
                  type="button"
                >
                  <button
                    onClick={[MockFunction]}
                    type="button"
                  >
                    Cancel
                  </button>
                </Component>
              </div>
            </Component>
          </div>
        </Component>
      </div>
    </Component>
  </FormControls>
</ContextWrapper>
`;

exports[`<FormControls /> should render only submit button 1`] = `
<ContextWrapper>
  <FormControls
    canReset={false}
    canSubmit={false}
    cancelLabel="Cancel"
    onReset={[MockFunction]}
    onSubmit={[MockFunction]}
    pristine={true}
    resetLabel="Reset"
    submitLabel="Submit"
  >
    <Component
      xs={12}
    >
      <div
        xs={12}
      >
        <Component>
          <div>
            <Component>
              <div>
                <Component
                  label="Submit"
                  onClick={[MockFunction]}
                  type="button"
                  variant="primary"
                >
                  <button
                    onClick={[MockFunction]}
                    type="button"
                    variant="primary"
                  >
                    Submit
                  </button>
                </Component>
              </div>
            </Component>
          </div>
        </Component>
      </div>
    </Component>
  </FormControls>
</ContextWrapper>
`;
